.circleci/config.yml

```diff
@@ -724,6 +725,136 @@ jobs:
           path: ./reports/junit
 
   # -------------------------
+  #     JOBS: iOS E2E Tests
+  # -------------------------
+  test_e2e_ios:
+    executor: reactnativeios
+    parameters:
+      ruby_version:
+        default: "2.7.7"
+        description: The version of ruby that must be used
+        type: string
+    steps:
+      - checkout_code_with_cache
+      - run_yarn
+      - attach_workspace:
+          at: .
+      - run:
+          name: Install appium
+          command: npm install appium@2.0.0-rc.4 -g
+      - run:
+          name: Install appium drivers
+          command: |
+            appium driver install uiautomator2
+            appium driver install xcuitest
+      - run:
+          name: Start Appium server
+          command: appium --base-path /wd/hub
+          background: true
+      - run:
+          name: Start Metro
+          command: |
+            cd packages/rn-tester
+            yarn start
+          background: true
+      - brew_install:
+          package: cmake
+      - setup_ruby:
+          ruby_version: << parameters.ruby_version >>
+      - run: 
+          name: Install Bundler
+          command: |
+            cd packages/rn-tester
+            bundle check || bundle install
+            bundle exec pod setup
+            bundle exec pod install --verbose
+      - run:
+          name: Boot iOS Simulator
+          command: source scripts/.tests.env && xcrun simctl boot "$IOS_DEVICE" || true
+      - run:
+          name: Build app
+          command: |
+              xcodebuild build \
+                -workspace packages/rn-tester/RNTesterPods.xcworkspace \
+                -configuration Debug \
+                -scheme RNTester \
+                -sdk iphonesimulator \
+                -derivedDataPath /tmp/e2e/
+      - run:
+          name: Move app to correct directory
+          command: mv /tmp/e2e/Build/Products/Debug-iphonesimulator/RNTester.app packages/rn-tester-e2e/apps/rn-tester.app
+      - run:
+          name: Check Appium server status
+          command: |
+            if ! nc -z 127.0.0.1 4723; then
+              echo Could not find Appium server!
+              exit 1
+            fi
+      - run:
+          name: Run E2E tests
+          command: | 
+            cd packages/rn-tester-e2e 
+            yarn test-ios-e2e
+
+  # -------------------------
+  #     JOBS: Android E2E Tests
+  # -------------------------
+  test_e2e_android:
+    executor:
+        name: android/android-machine
+        tag: 2023.07.1
+    steps:
+      - checkout_code_with_cache
+      - run_yarn   
+      - android/create-avd:
+          avd-name: e2e_emulator
+          system-image: system-images;android-33;google_apis;x86_64
```

Comment:
Not sure why we're installing `x86_64` but I believe we should be using `x86` here instead (according to CircleCI docs):
https://circleci.com/docs/android-machine-image/

Reply:
[it fails if we use `x86`](https://app.circleci.com/pipelines/github/facebook/react-native/27861/workflows/895f30e1-7ac5-4701-84f3-4ca778a40965/jobs/859070), we're keeping `x86_64 `

Reply:
Oh I see. It's because Google is not distributing system images for `x86` for API level >= 30 anymore. So let's stay with `x86_64` for now.